nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -627,7 +639,11 @@ in {
         Group = "matrix-synapse";
         WorkingDirectory = cfg.dataDir;
         PermissionsStartOnly = true;
-        ExecStart = "${cfg.package}/bin/homeserver --config-path ${configFile} --keys-directory ${cfg.dataDir}";
+        ExecStart = ''
+          ${cfg.package}/bin/homeserver \
+            --config-path ${ concatStringsSep " \\\n  --config-path " ([ configFile ] ++ cfg.extraConfigFiles) } \
```

Comment:
Hm, can you show me the desired expansion here? Just eyeballing this, it seems off to me. What about
```nix
lib.concatMapStringsSep " " (x: "--config-path ${x}")
```

Reply:
Yep sure, here is a snipped from a machine which I did produce this way:

```
ExecStart=/nix/store/f4g8ym6pk6w3g8fb1wk3b4g75p1vfhhl-python2.7-matrix-synapse-0.25.1/bin/homeserver \
  --config-path /nix/store/nmnb1zmni95csqfljwidiyz5mwgn3r1c-homeserver.yaml \
  --config-path /run/keys/matrix-synapse-secret-config \
  --keys-directory /var/lib/matrix-synapse
```

Still, using the other function is a better approach, it would be more obvious for the reader, I'll update it based on `concatMapStringsSep`, hope I'll find some time for it today.